### PR TITLE
parse ctts atom implemented

### PR DESCRIPTION
**Pull request by nklhrstv**
Implementation of ctts atom parsing.
@gkatsev @gesinger @joeyparrish could you review this PR? I need this atom in my project.

### CODE REVIEW

**Comment by joeyparrish** [comment]
I'm not one of the maintainers of this project, so I won't be able to approve the PR.  Sorry for any confusion!

**Comment by joeyparrish** [comment]
I'm not deeply familiar with norms and preferred style in mux.js, and I haven't compared your parser to any spec for the ctts box, but the implementation looks reasonable to me, for what that's worth.

**Review by gkatsev — APPROVED** [review]
Style-wise, it matches existing code. Also, appreciate that there are tests! Appreciate you looking at this @joeyparrish.
I'll have to defer to @gesinger or @brandonocasey to do a deeper review.

**Comment by joeyparrish** [comment]
@gkatsev, if you are happy with the project-specific details, and it's just a question of correctness, I found docs from Apple detailing the ctts box, and the parser matches perfectly as far as I can tell:

https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/QTFFChap2/qtff2.html#//apple_ref/doc/uid/TP40000939-CH204-SW40

**Review by joeyparrish — APPROVED** [review]
Approved by an interested outsider, for whatever that's worth.  :-D

File: lib/tools/mp4-inspector.js
Comment (brandonocasey) [inline lib/tools/mp4-inspector.js:441]: in the spec for mp4 `version 0` of this box uses an `uint` while `version 1` uses an int. Would you mind making the change for this case?

**Comment by nklhrstv** [comment]
@joeyparrish Sorry, i thought you are a maintainer

File: lib/tools/mp4-inspector.js
Comment (nklhrstv) [reply lib/tools/mp4-inspector.js:441]: can you please give me a reference for ctts spec?
[bento4](https://github.com/axiomatic-systems/Bento4/blob/be15647cc9750489cb2972f630d099b6426f03a7/Source/C%2B%2B/Core/Ap4CttsAtom.cpp#L102) actually uses only uint and used to have checks for quick time support
as far as i understand negative offsets ware used only in quicktime and they ware never part of the spec

File: lib/tools/mp4-inspector.js
Comment (brandonocasey) [inline lib/tools/mp4-inspector.js:441]: The spec is closed/paid for so I can't really link a reference to it. Basically you have to check the box version for this box and use `view.getUint32` if it is version 0 and `view.getInt32` for version 1. Should be something like this suggestion.
```suggestion
          sampleOffset: view[(result.version === 0 ? 'getUint32' :'getInt32')](i + 4)
```

**Comment by gkatsev** [comment]
We definitely value @joeyparrish's opinion!

**Comment by joeyparrish** [comment]
> We definitely value @joeyparrish's opinion!

Then I've fooled you all!!  😈

File: lib/tools/mp4-inspector.js
Comment (nklhrstv) [reply lib/tools/mp4-inspector.js]: Done

**Review by brandonocasey — APPROVED** [review]

**Pull request merged**
